mmf/datasets/builders/visual7w/dataset.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import torch
+from mmf.common.sample import Sample
+from mmf.datasets.mmf_dataset import MMFDataset
+from torchvision.ops import box_iou
+
+
+class Visual7WDataset(MMFDataset):
+    def __init__(self, config, *args, dataset_name="visual7w", **kwargs):
+        super().__init__(dataset_name, config, *args, **kwargs)
+
+    def __getitem__(self, idx):
+
+        sample_info = self.annotation_db[idx]
+        features_data = self.features_db[idx]
+        multiple_choice_idx = torch.tensor(sample_info["mc_idx"])
+
+        # image_info_0 contains roi features computed on a given image.
+        boxes = torch.from_numpy(features_data["image_info_0"]["bbox"])
+        features = features_data["image_feature_0"]
+
+        num_boxes = features.size()[0]
+        global_feature = torch.sum(features, 0) / num_boxes
+        global_feature = global_feature.unsqueeze(0)
+        image_w = features_data["image_info_0"]["image_width"]
+        image_h = features_data["image_info_0"]["image_height"]
+
+        global_bounding_box = torch.tensor(
+            [[0, 0, image_w, image_h]], dtype=torch.float
+        )
+
+        features = torch.cat((global_feature, features), 0)
+        boxes = torch.cat((global_bounding_box, boxes), 0)
+        num_boxes = num_boxes + 1
```

Comment:
I am not sure if using this global feature actually helps with anything. If not we don't need to add that. Can you check with running one experiment without the global feature addition?